src/main/java/org/tikv/common/PDClient.java

```diff
@@ -811,43 +819,60 @@ public String toString() {
     }
   }
 
-  private Metapb.Region decodeRegion(Metapb.Region region) {
+  @Override
+  public Metapb.Region decodeRegion(Metapb.Region region) {
     final boolean isRawRegion = conf.isRawKVMode();
     Metapb.Region.Builder builder =
         Metapb.Region.newBuilder()
             .setId(region.getId())
             .setRegionEpoch(region.getRegionEpoch())
             .addAllPeers(region.getPeersList());
 
-    if (region.getStartKey().isEmpty() || isRawRegion) {
+    if (region.getStartKey().isEmpty() || (conf.getApiVersion().isV1() && isRawRegion)) {
       builder.setStartKey(region.getStartKey());
     } else {
-      if (!conf.isTest()) {
-        byte[] decodedStartKey = BytesCodec.readBytes(new CodecDataInput(region.getStartKey()));
-        builder.setStartKey(ByteString.copyFrom(decodedStartKey));
-      } else {
-        try {
-          byte[] decodedStartKey = BytesCodec.readBytes(new CodecDataInput(region.getStartKey()));
-          builder.setStartKey(ByteString.copyFrom(decodedStartKey));
-        } catch (Exception e) {
-          builder.setStartKey(region.getStartKey());
+      try {
+        ByteString decodedStartKey =
+            ByteString.copyFrom(BytesCodec.readBytes(new CodecDataInput(region.getStartKey())));
+        if (conf.getApiVersion().isV2()) {
```

Comment:
This method has so many if, can you explain in code comment what behavior do you want.
Otherwise, I don't know if where is something wrong